### DISCUSSION TITLE
Keyword spotting end2end example

### DISCUSSION THREAD

**Discussion by tsaijohnson** [Xilinx/finn#435] [Q&A]
Hi,

Is there any possible to release the keyword spotting end to end example like cnv and tfc?

**Comment by heborras** [comment]
Hi @tsaijohnson,
Could you maybe elaborate a bit concerning what end2end part you mean exactly? Are you talking about the training code or the build flow or a combined notebook similar to the `cnv_end2end_example` notebook?

An example on how to build the bitfile is here in finn-examples: https://github.com/Xilinx/finn-examples/tree/main/build/kws
As for the training code: This is currently still in flux, but we are planning to publish it in late January.

**Reply by tsaijohnson** [reply]
Hi @HenniOVP,
What I mean is a combined notebook similar to the [cnv_end2end_example notebook ](https://github.com/Xilinx/finn/blob/main/notebooks/end2end_example/bnn-pynq/cnv_end2end_example.ipynb)
